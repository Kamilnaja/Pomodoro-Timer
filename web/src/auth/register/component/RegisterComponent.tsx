import { useRef } from 'react';
import { useForm } from 'react-hook-form';
import { ValidationMessage } from 'shared/components/validationMessage/ValidationMessage';
import { Registration } from '../../../../../types/interfaces';
import { AuthState } from '../../store/interfaces/auth.state';

export interface RegisterProps {
  handleSubmit: (arg: Registration) => void;
  handleClose: () => void;
  formState: AuthState;
}

type FormData = {
  login: string;
  email: string;
  password: string;
  repeatedPassword: string;
};

export const RegisterComponent = (props: RegisterProps) => {
  const { register, errors, handleSubmit, watch } = useForm<FormData>();
  const password = useRef({});
  password.current = watch('password', '');

  const onSubmit = (data: Registration) => {
    props.handleSubmit(data);
  };

  return (
    <div className="register">
      {props.formState.isSuccess ? (
        <div>ðŸ¤©Registered successfully, now you can log in</div>
      ) : (
        <>
          <h2 className="register__header">Enter registration data</h2>
          {props.formState.error?.message && (
            <ValidationMessage type={'error'} message={props.formState.error.message} />
          )}
          <form className="register__form form" onSubmit={handleSubmit(onSubmit)}>
            <div className="form__row">
              <label className="form__label label" htmlFor="login">
                User login
              </label>
              <input
                className="form__input input"
                type="text"
                name="login"
                id="login"
                placeholder="John Doe"
                ref={register({ required: 'ðŸ˜±Login is required' })}
                required
                autoFocus
              />
              {errors.login && <ValidationMessage type={'error'} message={errors.login.message as string} />}
            </div>
            <div className="form__row">
              <label className="form__label label" htmlFor="email">
                Email
              </label>
              <input
                className="form__input input"
                type="text"
                placeholder="johndoe@gmail.com"
                name="email"
                id="email"
                ref={register({ required: true, pattern: /^\S+@\S+$/i })}
                required
              />
              {errors.email?.type === 'required' && (
                <ValidationMessage type={'error'} message={'ðŸ˜“Email is required'} />
              )}
              {errors.email?.type === 'pattern' && (
                <ValidationMessage type={'error'} message={'ðŸ¤¥Please provide correct email'} />
              )}
            </div>
            <div className="form__row">
              <label className="form__label label" htmlFor="password">
                Password
              </label>
              <input
                className="form_input input"
                name="password"
                id="password"
                type="password"
                ref={register({
                  required: 'ðŸ¤¯You must specify a password',
                  minLength: { value: 8, message: 'ðŸ˜±Password must have at least 8 characters' },
                })}
                required
              />
              {errors.password && <ValidationMessage type={'error'} message={errors.password.message as string} />}
            </div>
            <div className="form__row">
              <label className="form__label label" htmlFor="repeatedPassword">
                Repeat password
              </label>
              <input
                className="form__input input"
                name="repeatedPassword"
                id="repeatedPassword"
                type="password"
                ref={register({ validate: value => value === password.current || 'ðŸ˜¨The passwords do not match' })}
                required
              />
              {errors.repeatedPassword && (
                <ValidationMessage type={'error'} message={errors.repeatedPassword.message as string} />
              )}
            </div>
            <button className="form__button" value="WyÅ›lij" type="submit">
              Submit
            </button>
          </form>
        </>
      )}
    </div>
  );
};
