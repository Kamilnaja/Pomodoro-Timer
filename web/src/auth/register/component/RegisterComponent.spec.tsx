import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { createAuthState } from '../../testing/auth.testing.data';
import { RegisterComponent } from './RegisterComponent';

describe('RegisterComponent', () => {
  let registerComponent = shallow(<RegisterComponent handleSubmit={() => {}} formState={createAuthState()} />);

  it('Should match snapshot', () => {
    expect(shallowToJson(registerComponent)).toMatchSnapshot();
  });
});
