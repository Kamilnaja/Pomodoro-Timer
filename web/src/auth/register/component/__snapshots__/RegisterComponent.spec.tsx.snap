// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterComponent Should match snapshot 1`] = `
<div
  className="register"
>
  <div>
    ðŸ¤©Registered successfully, now you can log in
  </div>
</div>
`;
